zerver/lib/redis_utils.py

```diff
@@ -11,9 +11,10 @@ def get_redis_client() -> redis.StrictRedis:
 
 def put_dict_in_redis(redis_client: redis.StrictRedis, key_format: str,
                       data_to_store: Dict[str, Any],
-                      expiration_seconds: int) -> str:
+                      expiration_seconds: int,
+                      token_length: int=64) -> str:
     with redis_client.pipeline() as pipeline:
-        token = generate_random_token(64)
+        token = generate_random_token(token_length)
```

Comment:
We should probably at least add tests for the exception behavior being reasonable for invalid keys being passed to `redis` similar to what you did with memcached access recently.

Reply:
Seems like there is no such thing as invalid redis key - https://redis.io/topics/data-types-intro#redis-keys - anything is fine, up to 512MB of size, so I don't think we need to test anything here?

Reply:
I think it may still make sense for us to add our own validation that rejects keys above 1024 characters before sending to our redis library; we never intend to use keys that long, so anything involving them is probably a bug.  And I could imagine a DoS attack that involves sending requests to Zulip that causes us to try to look up a bunch of invalid 512MB keys.

(And I think an analogue of the memcached commit to check that these tokens have the right format and bail early if not, similar to the API key validation, would be wise as well, just for sanity's sake).

Reply:
Pushed two commits with validation of the keys inside the redis_utils functions. I think it makes sense to require passing the intended key format to the ``get_`` function to protect against callers forgetting to validate key on their own.